client/src/ecash/mod.rs

```diff
@@ -93,7 +93,7 @@ impl ClientEcashWallet {
     ) -> anyhow::Result<()> {
         let spending_conditions = Self::assemble_escrow_conditions(contract, escrow_registration)?;
         self.wallet
-            .verify_token_p2pk(&escrow_token, spending_conditions)?;
```

Comment:
this expression creates a reference which is immediately dereferenced by the compiler
for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow
`#[warn(clippy::needless_borrow)]` on by defaultclippy[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B0%5D?0#file%3A%2F%2F%2Fworkspaces%2Fcashu-escrow-kit%2Fclient%2Fsrc%2Fecash%2Fmod.rs)
mod.rs(96, 32): change this to: `escrow_token`